### PR TITLE
[OSD-3574] remove osdManagedAdminSRE user and secret

### PR DESCRIPTION
**Pull request by dastergon**
This PR removes the `osdManagedAdminSRE` user and its secrets since it's no longer needed in the operator.

PTAL

See: https://issues.redhat.com/browse/OSD-3574

### CODE REVIEW

**Comment by drpaneas** [comment]
This doesn't work for me when I'm running the test. I am getting this error:

```
# Create CCS Secret
./hack/scripts/aws/rotate_iam_access_keys.sh -p osd-staging-2 -u osdCcsAdmin -a 11111111111 -n test-ccs-namespace -o /dev/stdout | oc apply -f -

An error occurred (InvalidClientTokenId) when calling the AssumeRole operation: The security token included in the request is invalid.
error: unable to decode "STDIN": json: cannot unmarshal string into Go value of type unstructured.detector
make: *** [create-ccs-secret] Error 1
```

File: docs/2.2-Account.md [hunk 1]
```diff
@@ -23,19 +23,17 @@ The `Account` controller is triggered by creating or changing an `Account` CR. I
 If the **awsLimit** set in the constants is not exceeded:
 
 1. Creates a new account in the organization belonging to credentials in secret `aws-account-operator-credentials`
-2. Configures two AWS IAM users from `iamUserNameUHC` and `iamUserNameSRE` as their respective usernames
+2. Configures two AWS IAM users from `iamUserNameUHC` as their respective usernames
```
Comment (dkeohane) [inline docs/2.2-Account.md:26]: This doesn't make sense any more to talk about plural

File: test/integration/test_secrets.sh [hunk 2]
```diff
@@ -11,16 +11,12 @@ EXIT_STATUS=$EXIT_PASS
...
 EXPECTED_SECRETS=(
-  "osdmanagedadminsre-secret:OSDMASRE_SECRET_KEYS"
   "secret:SECRET_KEYS"
 )
 
-OSDMASRE_SECRET_KEYS="aws_access_key_id aws_secret_access_key aws_user_name"
 SECRET_KEYS="aws_access_key_id aws_secret_access_key aws_user_name"
 
 for secret_map in "${EXPECTED_SECRETS[@]}"; do
-  test_secret_validity=false
```
Comment (dkeohane) [inline test/integration/test_secrets.sh:22]: Can you comment on what this does? Or it's not being used to you're removing it?

Comment (dastergon) [reply test/integration/test_secrets.sh:22]: not used

**Review by drpaneas — CHANGES_REQUESTED** [review]
The CI seems red:

```
pkg/controller/account/account_controller.go:325:20: undefined: iamUserNameSRE
pkg/controller/account/account_controller.go:331:38: undefined: iamUserNameSRE
pkg/controller/account/iam.go:444:38: undefined: iamUserNameSRE
```

**Review by drpaneas — APPROVED** [review]
Now it works, thanks for working on this Pavlos
/lgtm

**Comment by lisa** [comment]
/lgtm
/approve

love to see docs updated 💯

**Pull request merged**